core/src/main/java/org/jclouds/http/internal/JavaUrlHttpCommandExecutorService.java

```diff
@@ -72,6 +74,7 @@
    protected Supplier<SSLContext> sslContextSupplier;
    protected final int outputSocketBufferSize;
    protected final String userAgent;
+   protected final AtomicReference<SSLSocketFactory> sslSocketFactoryReference;
```

Comment:
I'm a bit unclear on this state on a singleton object. Could we just push back the decision on the SSL socket factory to use to the existing suppliers? If the suppliers are returning a singleton instance, as it is the case of the untrusted supplier, then we should be good here?
Instead of introducing state here, I'd just delegate that to the suppliers, as that is something that can be tweaked by users bu configuring their own ones when creating the context, if needed.

Reply:
I don't think I can use the existing suppliers to return a singleton socket factory, SSLContext is not an interface I can easily implement. I could change all Supplier<SSLContext>s to Supplier<SSLSocketFactory>s, that way the supplier implementation could contain the state.

Reply:
My main concern here is that the `JavaUrlHttpCommandExecutorService` is a singleton, and setting a member variable once will set its value for the entire lifecycle of the jclouds context. That means, every HTTP connection to any API method. Is this the intended behavior?

Reply:
Yeah that is intended - similarly to the way HttpsURLConnection uses the same instance of the default SSLSocketFactory (provided by SSLSocketFactory.getDefault()) in the case in which its SSLSocketFactory has not been explicitly set. Using the same SSLSocketFactory instance will allow HttpsClient to reuse existing connections from its KeepAliveCache.

I think that the better solution would be to inject Supplier<SSLSocketFactory>.